src/adapter/src/catalog/storage.rs

```diff
@@ -358,6 +372,27 @@ async fn migrate(
             txn.cluster_replicas.update(|_k, v| Some(v.clone()))?;
             Ok(())
         },
+        // Attributes were added to role definitions.
+        //
+        // Introduced in v0.45.0
+        //
+        // TODO(jkosh44) Can be cleared (patched to be empty) in v0.46.0
+        |txn: &mut Transaction<'_>, _now, _bootstrap_args| {
+            txn.roles.update(|_role_key, role_value| {
+                let mut role_value = role_value.clone();
+                let is_materialize_role = role_value.role.name == MATERIALIZE_ROLE;
+                let is_mz_introspection_role = role_value.role.name == INTROSPECTION_USER.name;
+                role_value.role.attributes.super_user = !is_mz_introspection_role;
+                role_value.role.attributes.inherit = true;
+                role_value.role.attributes.create_role = is_materialize_role;
+                role_value.role.attributes.create_db = is_materialize_role;
```

Comment:
Why do these need to be true for the mz role? If it's a superuser then can we always make these false, including the initial migration above? I think there's a few places where the RoleAttributes struct gets made, and it might be clearer if we used a constructor with whatever params we need to do things consistently.

Reply:
This is leftover from when I was going to use `materialize` as the default admin role. Though I still think this should be the case for `mz_system`. It's a bit redundant to have these be true and have `super_user` be true, but I was trying to match PostgreSQL's behavior here. They set everything to be true for the default role:
```
postgres=# SELECT * FROM pg_authid WHERE rolname = 'postgres';
 oid | rolname  | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolbypassrls | rolconnlimit | rolpassword | rolvaliduntil 
-----+----------+----------+------------+---------------+-------------+-------------+----------------+--------------+--------------+-------------+---------------
  10 | postgres | t        | t          | t             | t           | t           | t              | t            |           -1 |             | 
(1 row)
```

Reply:
What type of params were you thinking the constructor would take? I had the same though but struggled to think of a design that worked in all contexts.

Reply:
Ok, I actually think I came up with a clean constructor approach. Let me know what you think.